plugin/entryAttributes.go

```diff
@@ -285,13 +285,11 @@ func (a *EntryAttributes) UnmarshalJSON(data []byte) error {
 		a.SetCrtime(t)
 	}
 	if mode, ok := mp["mode"]; ok {
-		// Even though os.FileModes are uint32 types, json.Unmarshal unmarshals them as float64.
-		// That's ok, because float64 has sufficient precision to represent all uint32 types.
-		if raw, ok := mode.(float64); ok {
-			a.SetMode(os.FileMode(raw))
-		} else {
-			return attrMungeError("mode", fmt.Errorf("mode was unexpected type %T: %v", mode, mode))
+		md, err := munge.ToUintMode(mode)
```

Comment:
@MikaelSmith I think this should still preserve what you intended in https://github.com/puppetlabs/wash/commit/c7a7d43262fb2d98aecf4144f4e62a81234e48f8

Reply:
Or just change `munge.ToFileMode` to return `os.FileMode(<whatever ToUintMode returns>)`

Reply:
The bit representation of os.FileMode is not the same as that returned by stat. So this is incorrect. See the commit message for https://github.com/puppetlabs/wash/commit/c7a7d43262fb2d98aecf4144f4e62a81234e48f8#diff-db372dbae47f5fb2b6db0b72a78758c4.

Reply:
I don't understand the issue. The previous code took a `float64` mode and passed in to `os.FileMode` as-is (and the reason makes sense). The current code uses `ToUintMode` to make sure that we also allow hexadecimal/octal modes.

Reply:
Ah, I must've been thinking about ToFileMode.